### PR TITLE
Oddly broken test.

### PR DESCRIPTION
**Pull request by aars**
This one test is broken in an unapparent fashion (for me at least).

This is the only test that breaks after I added my "cast id to string" code. Oddly enough it doesn't have much to do with my changes though, as presented here (cloned from master).

Simply changing the ids to strings (what they MUST be according to JSONAPI) somehow ends up calling `addRelationship` on the resource, and then it breaks.

Also, minor and very unrelated issue in payload, the `name` property should be part of `attributes`.

### CODE REVIEW

File: tests/unit/adapters/application-test.js [hunk 1]
```diff
@@ -116,7 +116,7 @@ test('#findRelated can be called with optional type for the resource', function
   let service = this.container.lookup('service:people');
   let supervisor = this.container.lookup('model:employee').create({
     type: 'supervisors',
-    id: 1000000,
+    id: '1000000',
     name: 'The Boss',
```
Comment (aars) [inline tests/unit/adapters/application-test.js]: Should be in key `attributes`.

**Comment by aars** [comment]
```
Uncaught TypeError: Cannot read property 'cacheLookup' of undefined(…)
addRelationship @ resource.js:181
_updateRelationshipsData @ resource.js:123
didResolveProxyRelation @ resource.js:336
(anonymous function) @ related-proxy.js:105
tryCatch @ ember.debug.js:52150
invokeCallback @ ember.debug.js:
52165(anonymous function) @ ember.debug.js:54036
(anonymous function) @ ember.debug.js:41257
invoke @ ember.debug.js:990
flush @ ember.debug.js:1054
flush @ ember.debug.js:862
end @ ember.debug.js:176
(anonymous function) @ ember.debug.js:742
```

File: tests/unit/adapters/application-test.js [hunk 2]
```diff
@@ -116,7 +116,7 @@ test('#findRelated can be called with optional type for the resource', function
   let service = this.container.lookup('service:people');
   let supervisor = this.container.lookup('model:employee').create({
     type: 'supervisors',
```
Comment (pixelhandler) [inline tests/unit/adapters/application-test.js]: This should be `people`, see the example resource prototypes at https://github.com/pixelhandler/ember-jsonapi-resources/blob/master/tests/helpers/resources.js#L40-L48

**Comment by pixelhandler** [comment]
@aars this code was hiding the flaw in this breaking test, https://github.com/pixelhandler/ember-jsonapi-resources/blob/master/addon/models/resource.js#L136-L138

**Comment by pixelhandler** [comment]
@aars there is a hook that updates the relation after the related proxy resolves.

> Simply changing the ids to strings (what they MUST be according to JSONAPI) somehow ends up calling addRelationship on the resource, and then it breaks.

So yeah `addRelationship` updates the local data on the resource and uses a `cacheLookup` which wasn't stubbed in the adapter/service mock used in the test.

I'll create a PR to mixup that test.

**Comment by pixelhandler** [comment]
@aars I addressed this buggy test in PR #94

**Pull request closed without merging**